arelle/plugin/validate/DBA/rules/tr.py

```diff
@@ -192,6 +192,33 @@ def rule_tr09(
         )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=MULTI_TARGET_DISCLOSURE_SYSTEMS,
+)
+def rule_tr15(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.TR15: If the context identifier scheme is http://standards.iso.org/iso/17442, all context identifiers must be equal to the value of the field gsd:LegalEntityIdentifierOfReportingEntity.
+    """
+    contextEntityIdentifiers = {context.entityIdentifier for context in val.modelXbrl.contexts.values()}
+    leiFacts = val.modelXbrl.factsByQname.get(pluginData.legalEntityIdentifierOfReportingEntityQn, set())
+    if len(leiFacts) > 0:
+        leiFact = next(iter(leiFacts))
+        if leiFact.context is not None and leiFact.context.entityIdentifier[0] == 'http://standards.iso.org/iso/17442' and len(contextEntityIdentifiers) >0:
```

Comment:
This is checking all of the context entity identifiers. So this duplicates TR09. I would just check that the context on the lei fact matches the fact value. And then that lets TR09 does its thing.

Reply:
This line should be something like:
```suggestion
        if leiFact.context is not None and leiFact.context.entityIdentifier[0] == 'http://standards.iso.org/iso/17442' and leiFact.xValid >= VALID and leiFact.context.entityIdentifier[1] != leiFact.xValue:
```

This allows you to remove all of the `contextEnitytIdentifier` variable stuff.